### ISSUE TITLE
Deletion of mails and marking read or unread doesn't work when using Mailfence IMAP

### ISSUE THREAD

**Issue by markokocic** [thunderbird/thunderbird-android#6131]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.000

### Where did you get the app from?

Google Play

### Android version

11

### Device model

OnePlus Nord CE 5G

### Steps to reproduce

1. go to inbox
2. select multiple email messages
3. click delete & confirm deletion if prompted
4. observe that selected messages are not visible in inbox any more
5. refresh inbox by pulling down

### Expected behavior

All previously selected messages are deleted and don't re-appear after inbox refresh.

### Actual behavior

Previously deleted messages reappears in inbox.

Refreshing the inbox multiple times doesn't let the disappear. The only way is to delete them one by one.

**Comment by markokocic** [comment]
Looks like it's not only bulk deletion, but also marking email as read, marking mail as unread and staring of single mails that doesn't work.

Mail provider used is mailfence.com using IMAP protocol.

It's not an issue of mail provider, since I verified that all those operations work correctly using Fair Mail mobile client, Claws desktop mail client and using Mailfence own web interface.

**Comment by cketti** [comment]
Please record a debug log while trying to delete multiple messages. See https://github.com/thundernest/k-9/wiki/LoggingErrors

**Comment by markokocic** [comment]
hi @cketti, thanks for looking up.

As mentioned, it's not only deleting that doesn't work. Here's a quick summary, what works and what doesn't:

- Sending message - works
- Moving message to some folder - works
- Mark message as read / unread - doesn't work
- Deleting message - doesn't work
- Moving message to trash folder - this one is even stranger. It doesn't actually move the message to trash on server, but when refreshing the inbox in K-9, the message doesn't reappear again either.

What I can confirm is that I didn't have any of these issue with FairEmail and BlueMail on Android, and Thunderbird and Claws desktop email clients. I don't know what they do differently than K-9.

So far K-9 is the only client that have issue with Mailfence IMAP. If you have time to take look at other issue, I will try to contact Mailfence support and point them to this issue.

**Comment by cketti** [comment]
The server is behaving in unexpected ways. There's no reason to believe the other cases are fundamentally different. 

What K-9 Mail does, looks correct to me. And from what I can tell, it works fine with all other email providers. There's no point in speculating why K-9 Mail is triggering this behavior and other clients don't. The sequence of commands K-9 Mail uses should work, but doesn't. Please contact your email provider to find out what's going on.

**Comment by factome** [comment]
@markokocic I am experiencing the same problem, did you find a solution?

I contacted mailfence support and they said it may be throttling. I switched "Erase deleted messages from server" from "immediate" to "when polling". We'll see if this has an effect.

**Comment by markokocic** [comment]
@cketti is right, Mailfence has some stability issues.

After reporting this, I also started experiencing a different issue with Fair Email. There, Mailfence was not updating message list and status properly using existing IMAP connection. Instead, you needed to force disconnect and reconnect to their IMAP server to get updates in the Inbox.

Since I was getting random (different) errors in both K-9 and Fair Email, I fixed the problem after a week by switching to another mail provider. Since the switch, both K-9 and FairEmail work properly.

I am closing the issue, since I can confirm that it's not K-9 problem.
